tests/gpu-tests/test_train.py

```diff
@@ -140,3 +140,42 @@ def test_dpo():
     )
     # only checking the total, since model is tiny
     assert metrics['num_entries'] == 10
+
+
+@pytest.mark.gpu
+def test_rm():
+    model_path = os.getenv('NEMO_SKILLS_TEST_NEMO_MODEL')
+    if not model_path:
+        pytest.skip("Define NEMO_SKILLS_TEST_NEMO_MODEL to run this test")
+
+    model_path = os.getenv('NEMO_SKILLS_TEST_NEMO_MODEL')
+    if not model_path:
+        pytest.skip("Define NEMO_SKILLS_TEST_NEMO_MODEL to run this test")
+
+    train(
+        ctx=wrap_arguments(
+            "++trainer.rm.val_check_interval=1 "
+            "++trainer.rm.save_interval=1 "
+            "++trainer.rm.limit_val_batches=1 "
+            "++trainer.rm.max_steps=3 "
+            "++trainer.rm.max_epochs=10 "
+            "++model.data.train_ds.add_eos=False "
+            "++model.global_batch_size=2 "
+            "++model.micro_batch_size=1 "
+            "++model.optim.lr=1e-6 "
+            "++model.optim.sched.warmup_steps=0 "
+            "++model.tensor_model_parallel_size=1 "
+            "++model.pipeline_model_parallel_size=1 "
+        ),
+        cluster="test-local",
+        config_dir=Path(__file__).absolute().parent,
+        expname="test-rm",
+        training_algo="rm",
+        output_dir="/tmp/nemo-skills-tests/test-rm",
+        nemo_model=model_path,
+        num_nodes=1,
+        num_gpus=1,
+        num_training_jobs=1,
+        training_data="/nemo_run/code/tests/data/small-rm-data.test",
+        disable_wandb=True,
+    )
```

Comment:
Can you add some assert that checkpoints are created (which we can later replace with an inference call when that's integrated)? The problem is that nemo-run unfortunately always returns 0 exit code from any commands even if they crashed and so this train function will never cause the test to fail